modules/local/ctatsplicing/startocancerintrons/main.nf

```diff
@@ -0,0 +1,72 @@
+process CTATSPLICING_STARTOCANCERINTRONS {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://data.broadinstitute.org/Trinity/CTAT_SINGULARITY/CTAT-SPLICING/ctat_splicing.v0.0.2.simg' :
+        'docker.io/trinityctat/ctat_splicing:0.0.2' }"
+
+    input:
+    tuple val(meta), path(split_junction), path(junction), path(bam), path(bai)
+    tuple val(meta2), path(genome_lib)
+
+    output:
+    tuple val(meta), path("*.cancer_intron_reads.sorted.bam")       , emit: cancer_introns_sorted_bam
+    tuple val(meta), path("*.cancer_intron_reads.sorted.bam.bai")   , emit: cancer_introns_sorted_bai
+    tuple val(meta), path("*.gene_reads.sorted.sifted.bam")         , emit: gene_reads_sorted_bam
+    tuple val(meta), path("*.gene_reads.sorted.sifted.bam.bai")     , emit: gene_reads_sorted_bai
+    tuple val(meta), path("*.cancer.introns")                       , emit: cancer_introns
+    tuple val(meta), path("*.cancer.introns.prelim")                , emit: cancer_introns_prelim
+    tuple val(meta), path("*${prefix}.introns")                     , emit: introns
+    tuple val(meta), path("*.introns.for_IGV.bed")                  , emit: introns_igv_bed, optional: true
+    tuple val(meta), path("*.ctat-splicing.igv.html")               , emit: igv_html, optional: true
+    tuple val(meta), path("*.igv.tracks")                           , emit: igv_tracks, optional: true
+    tuple val(meta), path("*.chckpts")                              , emit: chckpts
+    path "versions.yml"                                             , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def bam_arg = bam ? "--bam_file ${bam}" : ""
+    def VERSION = '0.0.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def create_index = bam && !bai ? "samtools index ${bam}" : ""
```

Comment:
It was mainly a convenience thing because it wasn't available before, but you can decide if you think it would be better to separate it

Reply:
I think it would be better to separate it, to keep modules as minimal as possible

Reply:
This might be something to do when you separate the star align module from the other subworkflows (I don't want to do any work that will get changed any way). I can help with this if you want. This change would also make it so that CTAT splicing will only have to be run once per sample I think

Reply:
That sounds good, I will ask for your input on the merging of the aligment steps once I get there

Reply:
Thank you!